utils/writeable/src/lib.rs

```diff
@@ -249,6 +249,23 @@ pub trait Writeable {
     }
 }
 
+/// Implements [`Display`](core::fmt::Display) for types that implement [`Writeable`].
+///
+/// It's recommended to do this for every [`Writeable`] type, as it will add
+/// support for `core::fmt` features like [`fmt!`](std::fmt),
+/// [`print!`](std::print), [`write!`](std::write), etc.
+#[macro_export]
+macro_rules! impl_display_with_writeable {
+    ($type:ty) => {
+        impl core::fmt::Display for $type {
+            #[inline]
+            fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
```

Comment:
This won't show up on the `to_string` method though...

Reply:
(that comes from a blanket `impl<T: Display> ToString for T`)

Reply:
I know that, but it's the closest we can get and it's better than nothing. One reason I'm lukewarm about implementing `Display` is because it adds `ToString` which is a strict downgrade over what we can do with `write_to_string`.

Reply:
It's not actually strict, for example in datetime none of the formatted types implement `length_hint`, so those methods are actually the same.

Reply:
Acknowledged. There should be a modest code size improvement too by avoiding the need to go through `core::fmt::Formatter`.